apps/wwaw/build_wwaw.py

```diff
@@ -0,0 +1,492 @@
+# It extracts page content, builds agents with instructional context, enforces maximum fan-out rules,
+# and outputs a HOCON-formatted agent registry file for integration into a multi-agent system.
+#
+# Optional command-line arguments:
+# --total_agents: Number of agents to generate (default: TOTAL_AGENTS)
+# --start_url: Starting point for the crawl (default: START_URL)
+# --agent_network_name: Name for the generated agent network (default: AGENT_NETWORK_NAME)
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Generate a hierarchy of web agents.")
+    parser.add_argument("--total_agents", type=int, required=False,
+                        help=f"Total number of agents to generate (default {TOTAL_AGENTS})")
+    parser.add_argument("--start_url", type=str, required=False,
+                        help=f"Starting URL (default {START_URL})")
+    parser.add_argument("--agent_network_name", type=str, required=False,
+                        help=f"Agent network name (default: {AGENT_NETWORK_NAME})")
+
+    args = parser.parse_args()
+
+    # Use defaults if args are missing
+    the_start_url = args.start_url or START_URL
+    the_total_agents = args.total_agents or TOTAL_AGENTS
+    the_agent_network_name = args.agent_network_name or AGENT_NETWORK_NAME
+    the_agents = crawl(the_start_url, the_total_agents)
+    the_agents = enforce_fanout_recursive(the_agents, max_children=MAX_CHILDREN)
+    the_linked = set()
+    for an_agnt in the_agents.values():
+        the_linked.update(an_agnt["down_chains"])
+
+    # 1) single, safe hub insertion
+    the_unlinked = [
+        a for a in the_agents
+        if a not in the_linked and the_agents[a]["top_agent"] == "false"
+    ]
+    if the_unlinked:
+        the_hub_name = "unlinked_hub"
+        # only create once
+        if the_hub_name not in the_agents:
+            add_agent(
+                the_agents,
+                the_hub_name,
+                "You connect otherwise unlinked agents.",
+                the_unlinked,
+                "false"
+            )
+
+        # pick the real top (excluding the hub itself)
+        candidates = {
+            name: data
+            for name, data in the_agents.items()
+            if name != the_hub_name
+        }
+        the_top = max(candidates.items(), key=lambda x: len(x[1]["down_chains"]))[0]
+        # guard against self-reference and double-appending
+        if the_hub_name != the_top and the_hub_name not in the_agents[the_top]["down_chains"]:
+            the_agents[the_top]["down_chains"].append(the_hub_name)
+
+    # 2) FINAL SANITY: strip any remaining self-references
+    for name, data in the_agents.items():
+        data["down_chains"] = [child for child in data["down_chains"] if child != name]
+
+    # now generate HOCON
+    hocon = get_agent_network_hocon(the_agents, the_agent_network_name)
+
+    # Write the agent network file
+    file_path = OUTPUT_PATH + the_agent_network_name + ".hocon"
```

Comment:
Windows will have a problem with this. Use Path(blah blah) and then you can use file_path.open() below and Windows will be happier.